beta/src/types.ts

```diff
@@ -0,0 +1,88 @@
+import type { Stripe } from '@stripe/stripe-js'
+import type { IState } from 'choo'
+import Component from 'choo/component'
+import type { Track } from './stores/tracks.types'
+
+// The following two types are utility types that remove the index keys from a type (e.g. [key: string]: any)
+// Source: https://stackoverflow.com/a/51955852/998919
+type RemoveIndex<T> = {
+  [ K in keyof T as string extends K ? never : number extends K ? never : K ]: T[K]
+}
+type KnownKeys<T> = keyof RemoveIndex<T>
+
+// Utility type for defining that you want a class, rather than an instance of a class
+type Class<T> = new (...args: any[]) => T
+
+export interface AppState extends Pick<IState, KnownKeys<IState>> {
```

Comment:
I did my best in speccing this out. It's gonna have to be a WIP that will improve as more and more code gets brought into TS-land. Theoretically, it'll correct itself the more types we add (or bugs/irrational behavior will be exposed).